python/utils.py

```diff
@@ -211,6 +211,19 @@ def get_total_num(outputs):
         n = n + np.prod(sz)
     return n
 
+
+def get_total_num_mask(masks, props=None):
+    '''Returns the total number of active voxels in a forward pass'''
+    s = 0
+    for name, mask in masks.iteritems():
+        #full mask can correspond to empty array
+        if mask.size == 0 and props is not None:
+            s += props[name].size
+        else:
+            s += mask.sum()
```

Comment:
We do that already within the data provider, but that may be a bit safer if we end up changing the mask preprocessing. I'll commit a change for this